lib/flappi/response_builder.rb

```diff
@@ -184,7 +190,7 @@ def field(*args_or_name, block)
       def_args = extract_definition_args(args_or_name)
       require_arg def_args, :name
 
-      return if def_args.key?(:when) && !def_args[:when]
+      return unless check_when(def_args, @current_source)
```

Comment:
Shouldn't it return a `{}` instead of `nil` when the field is an :object?

I'm not sure how Kylor prefers our API. 
Ping @kylorhall. Our api should work: `{ instrument: {} }` or `{ instrument: nil }`?

Reply:
@felixibel @richdrich 
I would say `instrument: nil|null|undefined` is greatly preferred, if not a requirement when missing.

In most languages, checking the difference between an empty object `{}` and an instrument `{ id: …, code: … }` is non-trivial, eg. `if (instrument && instrument.id)` vs. just `if (instrument)`.

Reply:
Also goes the other way around in silly JS, like printing `instrument.name` becomes `instrument && instrument.name`. But I've seen we use lodash for such things now?

All good. Just wanted your green light for a `nil`

Reply:
@felixibel Not to sideline, unrelated to this PR:
1. We use `lodashGet` and `lodashSet` for the "path mechanism" exclusively, eg. `set({}, 'to.value', true)` = `{ to: { value: true } }`.  Basically, when we either need to get or set multiple levels at once (at times).  Often this is with variable or built paths.  You may see some `lodashGet(obj, 'simple')`, but that's just because in other places we call more complex paths.

2. I'm sure you know this, but yeah, rendering in JSX is commonly like `instrument && instrument.name`.  This is basically the same as:
```js
if (instrument) return instrument.name;
else return instrument;
```

- If `instrument` is false|undefined|null, it does not get rendered in JSX (if it's 0 it does).
- If it's truthy, it renders the name.